var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -220,6 +222,19 @@ def cmake_args(self):
             if "+rocm" in spec:
                 options.append(self.builder.define_hip_architectures(self))
 
+            # Add support for OpenMP with external clang
+            if spec.satisfies("%clang +openmp"):
+                clang = self.compiler.cc
+                clang_bin = os.path.dirname(clang)
+                clang_root = os.path.dirname(clang_bin)
+                libomp_ext = "dylib" if "platform=darwin" in spec else "so"
+                options.extend(
+                    [
+                        "-DOpenMP_CXX_LIB_NAMES=libomp",
+                        f"-DOpenMP_CXX_FLAGS={self.compiler.openmp_flag}",
+                        f"-DOpenMP_libomp_LIBRARY={clang_root}/lib/libomp.{libomp_ext}",
```

Comment:
@vicentebolea: Yes, SUSE and RHEL-based distro would have it in /usr/lib64, but I discovered now that is only needed to be set to any value to make the CMakeLists.txt file happy.

For example, on Ubuntu, the packages in the available apt repos are named `libomp-{7...15}-dev` and they install to `/usr/lib/llvm-15/lib/libomp.so`, which also does not match `"{clang_root}/lib/libomp.{libomp_ext}"`.

The `libomp-{7...15}-dev` are buggy in that installing one of them uninstalls any other libomp because all of them install their own `/usr/lib/x86_64-linux-gnu/libomp5.so`.

Since it is an extra package to know about and installing one version uninstall a previously installed version on Ubuntu distros, it looks like this distro-provided LLVM libomp should better not be used.

Adding `llvm-openmp` for clang builds (like I'd propose) takes care of spack installing libomp, but `CMakeLists.txt` still wants to have the variable set to something:

I tested that I can even pass `"-DOpenMP_libomp_LIBRARY=llvm-openmp"`, so I push a change to set it this way.

Side note on using `llvm-openmp` on Linux: While mostly depended on for `%apple-clang`, the spack recipes for the packages `survey` and `opensubdiv` already use `llvm-openmp` independent of the OS.

Reply:
Or better, we can set it like https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/onednn/package.py#L133:
```py
    f"-DOpenMP_libomp_LIBRARY={spec["llvm-openmp"].libs.libraries[0]}",
```